js/desktopCapturer/renderer.js

```diff
@@ -0,0 +1,265 @@
+            source.fileName = null;
+            applicationContent.appendChild(createItem(source));
+            hasApplications = true;
+        }
+    }
+
+    if (!hasScreens && !hasApplications) {
+        const errorContent = document.getElementById('error-content');
+        const mainContent = document.getElementById('main-content');
+
+        errorContent.style.display = 'block';
+        mainContent.style.display = 'none';
+    }
+
+    if (hasApplications) {
+        applicationTab.classList.remove('hidden');
+    }
+
+    if (hasScreens) {
+        screenTab.classList.remove('hidden');
+    }
+});
+
+function startShare() {
+    if (selectedSource && selectedSource.id) {
+        ipcRenderer.send('share-selected-source', selectedSource);
+        closeScreenPickerWindow();
+    }
+}
+
+/**
+ * Creates DOM elements and injects data
+ * @param source
+ * @returns {HTMLDivElement}
+ */
+function createItem(source) {
+    const itemContainer = document.createElement("div");
+    const sectionBox = document.createElement("div");
+    const imageTag = document.createElement("img");
+    const titleContainer = document.createElement("div");
+
+    // Added class names to the dom elements
+    itemContainer.classList.add('item-container');
+    sectionBox.classList.add('screen-section-box');
+    imageTag.classList.add('img-wrapper');
+    titleContainer.classList.add('screen-source-title');
+
+    // Inject data to the dom element
+    imageTag.src = source.thumbnail;
+    titleContainer.innerText = source.name;
+
+    sectionBox.appendChild(imageTag);
+    itemContainer.id = source.id;
+    itemContainer.appendChild(sectionBox);
+    itemContainer.appendChild(titleContainer);
+
+    itemContainer.addEventListener('click', updateUI.bind(this, source, itemContainer), false);
+
+    return itemContainer;
+}
+
+/**
+ * When ever user select a source store it and update the UI
+ * @param source
+ * @param itemContainer
+ */
+function updateUI(source, itemContainer) {
+    selectedSource = source;
+
+    let shareButton = document.getElementById('share');
+    shareButton.className = 'share-button';
+
+    highlightSelectedSource();
+    itemContainer.classList.add('selected');
+    shareButton.innerText = 'Share'
+}
+
+/**
+ * Loops through the items and removes
+ * the selected class property
+ */
+function highlightSelectedSource() {
+    let items = document.getElementsByClassName('item-container');
+    for (let i = 0, len = items.length; i < len; i++) {
+        items[i].classList.remove('selected');
+    }
+}
+
+/**
+ * Method that updates the share button
+ * text based on the content type
+ * @param text
+ */
+function updateShareButtonText(text) {
+    let shareButton = document.getElementById('share');
+
+    if (shareButton && shareButton.classList[0] === 'share-button-disable') {
+        shareButton.innerText = text;
+    }
+}
+
+/**
+ * Method handles used key up event
+ * @param e
+ */
+function handleKeyUpPress(e) {
```

Comment:
Could be renamed to `handleKeyboardInteractions`

Reply:
This method will be invoked only for `keyup` events